canard_can.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2016-2018 UAVCAN Team
+ *
+ * Distributed under the MIT License, available in the file LICENSE.
+ *
+ */
+
+#ifndef CANARD_CAN_H
+#define CANARD_CAN_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/// This will be changed when the support for CAN FD is added
+#define CANARD_CAN_FRAME_MAX_DATA_LEN               8U
```

Comment:
The comment is copy paste from previous implementation.

There are no static block size with my PR, as you point out, this would be very wasteful. Instead block size will be set when initializing the libcanard object depending on the transport protocol in use. 

The alternative would be to always use a small block size (like 32 bytes). This have some advantages. On the con side, it would require allocating several blocks for each outgoing frame.